expression/builtin_encryption.go

```diff
@@ -81,20 +103,40 @@ func (c *aesDecryptFunctionClass) getFunction(ctx sessionctx.Context, args []Exp
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(c.verifyArgs(args))
 	}
-	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, types.ETString, types.ETString)
+	argTps := make([]types.EvalType, 0, len(args))
+	for range args {
+		argTps = append(argTps, types.ETString)
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETString, argTps...)
 	bf.tp.Flen = args[0].GetType().Flen // At most.
 	types.SetBinChsClnFlag(bf.tp)
-	sig := &builtinAesDecryptSig{bf}
-	return sig, nil
+
+	blockMode, _ := ctx.GetSessionVars().GetSystemVar(variable.BlockEncryptionMode)
+	mode, exists := aesModes[strings.ToLower(blockMode)]
+	if !exists {
+		return nil, errors.Errorf("unsupported block encryption mode - %v", blockMode)
+	}
+	if mode.ivRequired {
+		if len(args) != 3 {
+			return nil, ErrIncorrectParameterCount.GenByArgs("aes_decrypt")
+		}
+		return &builtinAesDecryptIVSig{bf, mode}, nil
+	} else if len(args) == 3 {
+		// For modes that do not require init_vector, it is ignored and a warning is generated if it is specified.
+		ctx.GetSessionVars().StmtCtx.AppendWarning(errWarnOptionIgnored.GenByArgs("IV"))
```

Comment:
It seems this warning should be appended in the function execution phase for every input row:
```sql
MySQL(localhost:3306) > select aes_encrypt(a, b, c) from t;
+----------------------+
| aes_encrypt(a, b, c) |
+----------------------+
| 0SF&Qٜ               |
| 0SF&Qٜ               |
| 0SF&Qٜ               |
| 0SF&Qٜ               |
+----------------------+
4 rows in set, 4 warnings (0.00 sec)

MySQL(localhost:3306) > show warnings;
+---------+------+---------------------+
| Level   | Code | Message             |
+---------+------+---------------------+
| Warning | 1618 | <IV> option ignored |
| Warning | 1618 | <IV> option ignored |
| Warning | 1618 | <IV> option ignored |
| Warning | 1618 | <IV> option ignored |
+---------+------+---------------------+
4 rows in set (0.00 sec)
```

Reply:
moved to execution phase, and new testcase is added.